File: lib/models/group_v2_clan_info_and_investment.dart
import 'destiny_progression.dart';
import 'clan_banner.dart';
class GroupV2ClanInfoAndInvestment{
	Map<DestinyProgression, dynamic> d2ClanProgressions;
	String clanCallsign;
	ClanBanner clanBannerData;
	GroupV2ClanInfoAndInvestment(
		Map<DestinyProgression, dynamic> this.d2ClanProgressions,
		String this.clanCallsign,
		ClanBanner this.clanBannerData,
	);

	static GroupV2ClanInfoAndInvestment fromMap(Map<String, dynamic> data){
		if(data == null) {
			return null;
		};
		return new GroupV2ClanInfoAndInvestment(
				data['d2ClanProgressions'],
				data['clanCallsign'],
				ClanBanner.fromMap(data['clanBannerData']),
		);
	}

	static List<GroupV2ClanInfoAndInvestment> fromList(List<dynamic> data){
		if(data == null) {
			return null;
		};
		List<GroupV2ClanInfoAndInvestment> list = new List();
    data.forEach((item) {
      list.add(GroupV2ClanInfoAndInvestment.fromMap(item));
    });
    return list;
	}

	Map<String, dynamic> toMap(){
		Map<String, dynamic> data = new Map();
			data['d2ClanProgressions'] = d2ClanProgressions;
			data['clanCallsign'] = clanCallsign;
			data['clanBannerData'] = clanBannerData.toMap();
	}
}

File: BungieNetApi/Models/GroupV2ClanInfoAndInvestment.cs
using System.Runtime.Serialization;
using System.Collections.Generic;
using System.Linq;

namespace GhostSharper.Models
{
    /// <summary>
    /// The same as GroupV2ClanInfo, but includes any investment data.
    /// </summary>
    [DataContract]
    public class GroupV2ClanInfoAndInvestment
    {
        [DataMember(Name = "d2ClanProgressions", EmitDefaultValue = false)]
        public Dictionary<string, DestinyProgression> D2ClanProgressions { get; set; }

        [DataMember(Name = "clanCallsign", EmitDefaultValue = false)]
        public string ClanCallsign { get; set; }

        [DataMember(Name = "clanBannerData", EmitDefaultValue = false)]
        public ClanBanner ClanBannerData { get; set; }


        public override bool Equals(object input)
        {
            return this.Equals(input as GroupV2ClanInfoAndInvestment);
        }

        public bool Equals(GroupV2ClanInfoAndInvestment input)
        {
            if (input == null) return false;

            return
                (
                    D2ClanProgressions == input.D2ClanProgressions ||
                    (D2ClanProgressions != null && D2ClanProgressions.SequenceEqual(input.D2ClanProgressions))
                ) &&
                (
                    ClanCallsign == input.ClanCallsign ||
                    (ClanCallsign != null && ClanCallsign.Equals(input.ClanCallsign))
                ) &&
                (
                    ClanBannerData == input.ClanBannerData ||
                    (ClanBannerData != null && ClanBannerData.Equals(input.ClanBannerData))
                ) ;
        }
    }
}

